apps/sr/tid1500writer.cxx

```diff
@@ -190,22 +190,23 @@ int main(int argc, char** argv){
  }
 
   DSRDocument doc;
-  std::cout << "Setting tree from the report" << std::endl;
   OFCondition cond = doc.setTreeFromRootTemplate(report, OFTrue /*expandTree*/);
   if(cond.bad()){
     std::cout << "Failure: " << cond.text() << std::endl;
     return -1;
   }
 
   // cleanup duplicate modality from image descriptor entry
-  DSRDocumentTree &st = doc.getTree();
-  size_t nnid = st.gotoAnnotatedNode("TID 1601 - Row 1");
-  st.goDown();
-  st.removeSubTree();
-  while(nnid){
-    nnid = st.gotoNextAnnotatedNode("TID 1601 - Row 1");
-    st.goDown();
-    st.removeSubTree();
+  {
+    DSRDocumentTree &st = doc.getTree();
+    DSRCodedEntryValue modalityCode("121139","DCM","Modality");
+    size_t nnid = st.gotoAnnotatedNode("TID 1601 - Row 1");
```

Comment:
You don't check the returned node ID of gotoAnnotatedNode() but as I said: this is just a workaround and if it works for your use case... it's should be sufficient :)